data-prepper-core/src/main/java/org/opensearch/dataprepper/pipeline/server/ShutdownHandler.java

```diff
@@ -44,6 +50,7 @@ public void handle(final HttpExchange exchange) throws IOException {
             exchange.sendResponseHeaders(HttpURLConnection.HTTP_INTERNAL_ERROR, 0);
         } finally {
             exchange.getResponseBody().close();
+            Files.write(SHUTDOWN_FILE_PATH, SHUTDOWN_MESSAGE.getBytes(StandardCharsets.UTF_8));
             dataPrepper.shutdownServers();
```

Comment:
I chose to create the file before the shutdownServer call because otherwise there is a chance for the File write to be interrupted (since we are shutting down the server which is where the ShutdownHandler exists.